src/core/cgroup.c

```diff
@@ -3135,19 +3124,20 @@ int unit_get_memory_current(Unit *u, uint64_t *ret) {
         if (r < 0)
                 return r;
         if (r > 0)
-                r = cg_get_attribute("memory", u->cgroup_path, "memory.current", &v);
+                r = cg_get_attribute_as_uint64("memory", u->cgroup_path, "memory.current", &v);
         else
-                r = cg_get_attribute("memory", u->cgroup_path, "memory.usage_in_bytes", &v);
+                r = cg_get_attribute_as_uint64("memory", u->cgroup_path, "memory.usage_in_bytes", &v);
         if (r == -ENOENT)
                 return -ENODATA;
         if (r < 0)
                 return r;
 
-        return safe_atou64(v, ret);
+        *ret = v;
```

Comment:
With the helper, using `v` is not necessary, since `ret` can be passed to the helper. Assuming that ENODATA translation is folded into the helper, the last few lines can become:
```C
return cg_get_attribute_as_uint64("memory", u->cgroup_path,
                                         r > 0 ? "memory.current" : "memory.usage_in_bytes,
                                         ret);
```